packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -1034,10 +1034,12 @@ class Dropdown extends AutoControlledComponent<WithAsProp<DropdownProps>, Dropdo
   tryRemoveItemFromValue = () => {
     const { multiple } = this.props
     const { searchQuery, value } = this.state
+    const inputElement = this.inputRef.current
 
     if (
       multiple &&
-      (searchQuery === '' || this.inputRef.current.selectionStart === 0) &&
+      (searchQuery === '' ||
+        (inputElement.selectionStart === 0 && inputElement.selectionEnd === 0)) &&
```

Comment:
Better tests meant I could catch this bug. On search we should check that the cursor is at the beginning and not selecting any text in order to delete an element from selection.

Reply:
Can we extract bugfix to a separate PR?